File: mmosad19419-cs50-problems-2021-x-sentimental-readability/readability.py
import cs50

# define main function


def main():
    # Get Text from the user
    text = cs50.get_string("Text: ")

    # Count letters --> letters_c function
    letters = letters_c(text)

    # Count words --> words_c function
    words = words_c(text)

    # Count sentences --> sentences_c function
    sentences = sentences_c(text)

    # calculate l & s for index equ
    l = (letters * 100) / words
    s = (sentences * 100) / words

    # Calculate the index
    index = .0588 * l - .296 * s - 15.8
    # Calculate the grade
    grade = round(index)

    # Print the result
    if grade < 1:
        print("Before Grade 1...")
    elif grade > 16:
        print("Grade 16+")
    else:
        print("Grade {}".format(grade))


# Define letters count function


def letters_c(text):
    l = 0
    for i in range(len(text)):
        if text[i].isalpha():
            l += 1
    return l

# Define words count function


def words_c(text):
    w = 1
    for i in range(len(text)):
        if text[i] == " ":
            w += 1
    return w

# Define sentences count function


def sentences_c(text):
    s = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '?' or text[i] == '!':
            s += 1
    return s


if __name__ == "__main__":
    main()

File: mmosad19419-cs50-problems-2021-x-readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

//functions declarations
int count_l(string text);
int count_w(string text);
int count_s(string text);



int main(void)

{
//get the text input from the user
    string t = get_string("Text:");

//count the letters
    int letter = count_l(t);

//count the words
    int word = count_w(t);

// count the number of sentences
    int sentence = count_s(t);


// calculete L & S for the index equation
    double l = (letter * 100) / word;

    double s = (sentence * 100) / word;


//claculate the index & round it
    double index = 0.0588 * l - 0.296 * s - 15.8;

    index = round(index - .05);

    int grade = index;

//print the result
    if (grade < 1)
    {
        printf("Before Grade 1...\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf(" Grade %i\n", grade);
    }  
}




//count the letters (function)
int count_l(string text)
{
    int l = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            l++;
        }
    }
    return l;
}


//count the words (function)
int count_w(string text)
{
    int w = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            w++;
        }
    }
    return w;
}

// count the number of sentences (function)
int count_s(string text)
{
    int s = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            s++;
        }
    }
    return s;
}